### PR TITLE
Set required and aria on input when input is required

### LINKED ISSUE #604: If an input is required then the input does nog always get the required element
For example in the demo application email is a required field.  validates :email, presence: true

`<div class="form-group row"><label class="col-form-label col-sm-2 required" for="user_email">Email address</label><div class="col-sm-10"><input placeholder="Enter Email" class="form-control" type="email" name="user[email]" id="user_email"><small class="form-text text-muted">We'll never share your email with anyone else</small></div></div>`
Then the label has an element required but the input has not. This means that when a screenreader reads the form is is not mentioned that the input is required.

If an input got the option required:  true eq form.text_area :misc, required: true then the input field did get a required element

In BootstrapForm::FormGroupBuilder.form_group_builder_options and BootstrapForm::FormGroupBuilder.form_group_required. there is something done with required but it does not work how I expected it to work.

I added required and aria-required to all the input types, except the select.
Something in the BootstrapForm::Inputs::Select.select_with_bootstrap, from the form_group_builder it does get the options.
But when calling select_without_bootstrap(method, choices, options, html_options, &block) with the correct options it does not return required or aria-required.

I will add a PR with my changes

### PR DESCRIPTION
**Pull request by CharlieWinkwaves**
fixes #604

### CODE REVIEW

File: test/bootstrap_form_test.rb [hunk 1]
```diff
@@ -34,11 +34,11 @@ class BootstrapFormTest < ActionView::TestCase
           <div class="col-sm-10">
             <div class="form-check">
               <input class="form-check-input" id="user_misc_1" name="user[misc]" type="radio" value="1" />
-              <label class="form-check-label" for="user_misc_1"> Foo</label>
+              <label class="form-check-label" for="user_misc_1">Foo</label>
```
[the remark this answers is not in the document]
Comment (CharlieWinkwaves) [reply test/bootstrap_form_test.rb]: No problem fixed it what do you want me to do with the commit already merged to master?

**Review by lcreid — APPROVED** [review]
Thanks so much for this PR! Great work!

File: test/bootstrap_form_test.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (lcreid) [reply test/bootstrap_form_test.rb]: You could just write up an issue describing how Bootstrap 4 tests are different from Bootstrap 5, we can review it later. But if it's more work to do write up an issue than to do another PR for `main`, then submit another PR. Does that make sense?

[the remark this answers is not in the document]
Comment (donv) [reply test/bootstrap_form_test.rb]: This is resolved, right?

**Comment by donv** [comment]
This looks like a good change.  Any reason not to merge it?

@CharlieWinkwaves could you rebase onto master?

**Review by donv — APPROVED** [review]

File: test/bootstrap_form_test.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (CharlieWinkwaves) [reply test/bootstrap_form_test.rb]: Yes it is resolved

**Comment by CharlieWinkwaves** [comment]
I do not know why but is keeps saying that this PR has 4 commits in it

**Comment by CharlieWinkwaves** [comment]
@lcreid @donv We have pulled the latest v5 and we noticed that these changes are overwritten by a merge commit. And are thus not in the latest version

**Pull request closed without merging**